ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -92,6 +96,16 @@ data ImmDB m blk = ImmDB {
     , err          :: ErrorHandling ImmDB.ImmutableDBError m
     }
 
+instance MonadSTM m => NormalForm m (ImmDB m blk) where
+  custom ImmDB{..} = NF.checkMultiple
+    [ NF.custom immDB
+    , NF.ignore decBlock     -- Parameterised over s
+    , NF.ignore encBlock     -- A function
+    , NF.ignore immEpochInfo -- A record of functions
+    , NF.ignore isEBB        -- A function
+    , NF.ignore err          -- A record of functions
```

Comment:
Just curious: is it not worth asserting that all of these functions/records of functions are in normal form?

Reply:
Ah, while playing around in GHCi, I had the impression that functions never were in normal form.

The main reason for ignoring these is that they are static anyway, we never update them, so the chances are slim they cause space leaks.